doc: add a blog for Notation v1.2.0 release (#408)
notaryproject/notaryproject.dev: content/en/blog/2024/announcing-notation-v1-2.md

### Support OCI specification v1.1.0

In Feb 2024, the Open Container Initiative (OCI) community released version 1.1.0, which includes the [OCI image specification v1.1.0](https://github.com/opencontainers/image-spec/releases/tag/v1.1.0) and the [OCI distribution specification v1.1.0](https://github.com/opencontainers/distribution-spec/releases/tag/v1.1.0). Notation now adheres to the OCI spec v1.1.0, leading to the deprecation of the experimental flag `--allow-referrers-api`. A new flag, `--force-referrers-tag` (default to true), has been introduced to the `notation sign` command, enabling users to choose the referrers tag schema over the [referrers API](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#enabling-the-referrers-api) if the registry they are using does not yet support the referrers API. The `notation verify/list/inspect` commands will now attempt to use the referrers API first and will automatically fall back to the [referrers tag schema](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#referrers-tag-schema) if the referrers API is unsupported by the registry.

### Support for RFC 3161 compliant timestamps

> Since this release, Notary Project specifications and Notation tool supports RFC 3161 compliant Timestamps. Digital signatures must be generated within the certificate's validity period, as expired certificates compromise the signature's trustworthiness. Trusted timestamps extend the trust of signatures created within certificate validity, allowing successful signature verification even after certificates have expired. Learn more at the document [how to sign and verify artifacts in OCI-compliant registries using trusted timestamps](../../docs/user-guides/how-to/timestamps).

Reviewer: I think it worth mentioning Notation timestamp is built on the [tspclient-go library](https://github.com/notaryproject/tspclient-go) here.

> Since this release, Notary Project specifications and Notation tool supports RFC 3161 compliant Timestamps. Digital signatures must be generated within the certificate's validity period, as expired certificates compromise the signature's trustworthiness. Trusted timestamps extend the trust of signatures created within certificate validity, allowing successful signature verification even after certificates have expired. Learn more at the document [how to sign and verify artifacts in OCI-compliant registries using trusted timestamps](../../docs/user-guides/how-to/timestamps).

Reviewer 2: Since `Notary Project specifications` has a section itself, we don't need to repeat it here.
```suggestion
Since this release, Notation supports RFC 3161 compliant Timestamps. Digital signatures must be generated within the certificate's validity period, as expired certificates compromise the signature's trustworthiness. Trusted timestamps extend the trust of signatures created within certificate validity, allowing successful signature verification even after certificates have expired. Learn more at the document [how to sign and verify artifacts in OCI-compliant registries with timestamping](../../docs/user-guides/how-to/timestamps).
```

> Since this release, Notary Project specifications and Notation tool supports RFC 3161 compliant Timestamps. Digital signatures must be generated within the certificate's validity period, as expired certificates compromise the signature's trustworthiness. Trusted timestamps extend the trust of signatures created within certificate validity, allowing successful signature verification even after certificates have expired. Learn more at the document [how to sign and verify artifacts in OCI-compliant registries using trusted timestamps](../../docs/user-guides/how-to/timestamps).

Reviewer: According to the [RFC 3161](https://datatracker.ietf.org/doc/html/rfc3161.html) and [Notary Project Spec](https://github.com/notaryproject/specifications/blob/main/specs/signing-and-verification-workflow.md), is it much consistent to use `supports RFC 3161 compliant Timestamp standard (or protocol)` and `Trusted timestamping` in this paragraph? I seldomly see the term `timestamps`.

### Other changes

> Notation CLI now offers the `armv7` binary, enabling its usage in environments with arm32 processors.

Reviewer: ```suggestion
Notation CLI now offers the `armv7` binary, enabling its usage in environments with ARM32 processors.

```

## Get started with Notation v1.2.0

> You can follow the [quick start](https://notaryproject.dev/docs/quickstart/) to try Notation v1.2.0 for basic signing and verification workflow.

Reviewer: Suggest using a relative link `[quick start](../../docs/quickstart)` here. Otherwise, this link will always point to the latest version in the future.

## Deprecation

The experimental flag `--allow-referrers-api` used by `notation sign` and `notation verify` commands is now deprecated. See [Support OCI specification v1.1.0](#support-oci-specification-v110) for details.

## Notable Capabilities in this Release

Here are some of the major capabilities and features included in this release.

### Notary Project specifications

The Notary Project specifications now include support for [RFC 3161](https://www.rfc-editor.org/rfc/rfc3161) timestamping and introduce Notation plugin conventions in the [plugin specification](https://github.com/notaryproject/specifications/blob/v1.1.0/specs/plugin-extensibility.md).

### Support OCI specification v1.1.0

> In Feb 2024, the Open Container Initiative (OCI) community released version 1.1.0, which includes the [OCI image specification v1.1.0](https://github.com/opencontainers/image-spec/releases/tag/v1.1.0) and the [OCI distribution specification v1.1.0](https://github.com/opencontainers/distribution-spec/releases/tag/v1.1.0). Notation now adheres to the OCI spec v1.1.0, leading to the deprecation of the experimental flag `--allow-referrers-api`. A new flag, `--force-referrers-tag` (default to true), has been introduced to the `notation sign` command, enabling users to choose the referrers tag schema over the [referrers API](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#enabling-the-referrers-api) if the registry they are using does not yet support the referrers API. The `notation verify/list/inspect` commands attempt to use the referrers API first and automatically fall back to the [referrers tag schema](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#referrers-tag-schema) if the referrers API is not supported by the registry.

Reviewer 3: > A new flag, `--force-referrers-tag` (default to true)

You may want to mention that this flag will be default to `false` in future releases.

Reviewer 3: Examples of golang release notes can be found.

In [golang 1.22](https://go.dev/doc/go1.22) release notes:

> We plan to include an API migration tool in a future release, likely Go 1.23.

> We may change the default value of x509usepolicies in Go 1.23, making Policies the default field for marshaling.

> Go 1.23 will require macOS 11 Big Sur or later.

In [golang 1.23](https://go.dev/doc/go1.23) release notes:

> The [x509sha1 GODEBUG setting](https://go.dev/pkg/crypto/x509#InsecureAlgorithmError) will be removed in the next Go major release (Go 1.24). This will mean that crypto/x509 will no longer support verifying signatures on certificates that use SHA-1 based signature algorithms.

> Go 1.24 will require Linux kernel version 3.17 or later, with an exception that systems running 3.10 or later will continue to be supported if the kernel has been patched to support the getrandom system call.

> In Feb 2024, the Open Container Initiative (OCI) community released version 1.1.0, which includes the [OCI image specification v1.1.0](https://github.com/opencontainers/image-spec/releases/tag/v1.1.0) and the [OCI distribution specification v1.1.0](https://github.com/opencontainers/distribution-spec/releases/tag/v1.1.0). Notation now adheres to the OCI spec v1.1.0, leading to the deprecation of the experimental flag `--allow-referrers-api`. A new flag, `--force-referrers-tag` (default to true), has been introduced to the `notation sign` command, enabling users to choose the referrers tag schema over the [referrers API](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#enabling-the-referrers-api) if the registry they are using does not yet support the referrers API. The `notation verify/list/inspect` commands will now attempt to use the referrers API first and will automatically fall back to the [referrers tag schema](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#referrers-tag-schema) if the referrers API is unsupported by the registry.

Reviewer 2: nit:
```suggestion
In Feb 2024, the Open Container Initiative (OCI) community released version 1.1.0, which includes the [OCI image specification v1.1.0](https://github.com/opencontainers/image-spec/releases/tag/v1.1.0) and the [OCI distribution specification v1.1.0](https://github.com/opencontainers/distribution-spec/releases/tag/v1.1.0). Notation now adheres to the OCI spec v1.1.0, leading to the deprecation of the experimental flag `--allow-referrers-api`. A new flag, `--force-referrers-tag` (default to true), has been introduced to the `notation sign` command, enabling users to choose the referrers tag schema over the [referrers API](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#enabling-the-referrers-api) if the registry they are using does not yet support the referrers API. The `notation verify/list/inspect` commands now attempt to use the referrers API first and automatically fall back to the [referrers tag schema](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#referrers-tag-schema) if the referrers API is not supported by the registry.

```

### Support for RFC 3161 compliant timestamping

> Since this release, Notation supports RFC 3161 compliant timestamping. Digital signatures must be generated within the certificate's validity period, as expired certificates compromise the signature's trustworthiness. Timestamping extends the trust of signatures created within certificate validity, allowing successful signature verification even after certificates have expired. Notation's timestamping feature is built on top of the `[tspclient-go](https://github.com/notaryproject/tspclient-go)` library.

Reviewer: ```suggestion
Since this release, Notation supports RFC 3161 compliant timestamping. Digital signatures must be generated within the certificate's validity period, as expired certificates compromise the signature's trustworthiness. Timestamping extends the trust of signatures created within certificate validity, allowing successful signature verification even after certificates have expired. Notation's timestamping feature is built on top of the [tspclient-go](https://github.com/notaryproject/tspclient-go) library.

```

Reviewer: Nit: the hyperlink is not rendered as expected.

Learn more at the document [how to sign and verify artifacts in OCI-compliant registries with timestamping](../../docs/user-guides/how-to/timestamping).

### Other changes

> Notation CLI now offers the `armv7` binary, enabling its usage in environments with ARM32 processors.

Reviewer 3: It seems redundant to mention ARM32 processors. Is it for IoT scenarios?

## Deprecation

The experimental flag `--allow-referrers-api` used by `notation sign` and `notation verify` commands is now deprecated. See [Support OCI specification v1.1.0](#support-oci-specification-v110) for details.

## Notable Capabilities in this Release

Here are some of the major capabilities and features included in this release.

### Notary Project specifications

The Notary Project specifications now include support for [RFC 3161](https://www.rfc-editor.org/rfc/rfc3161) timestamping and introduce Notation plugin conventions in the [plugin specification](https://github.com/notaryproject/specifications/blob/v1.1.0/specs/plugin-extensibility.md).

### Support OCI specification v1.1.0

> In Feb 2024, the Open Container Initiative (OCI) community released version 1.1.0, which includes the [OCI image specification v1.1.0](https://github.com/opencontainers/image-spec/releases/tag/v1.1.0) and the [OCI distribution specification v1.1.0](https://github.com/opencontainers/distribution-spec/releases/tag/v1.1.0). Notation now adheres to the OCI spec v1.1.0, leading to the deprecation of the experimental flag `--allow-referrers-api`. A new flag, `--force-referrers-tag` (default to `true`), has been introduced to the `notation sign` command, enabling users to choose the referrers tag schema over the [referrers API](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#enabling-the-referrers-api) if the registry they are using does not yet support the referrers API. The `notation verify/list/inspect` commands attempt to use the referrers API first and automatically fall back to the [referrers tag schema](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#referrers-tag-schema) if the referrers API is not supported by the registry.

Author: If all registries support referrers API, then there is no need for adding this flag `--force-referrers-tag`.  We want referrers API to be the default one, this is the goal. However, not all registries currently support referrers API. Another reason is we do not want to introduce breaking changes in v1.2.0, since version v1.10 and prior versions by default use tag schema. 

In Notation v2.0, we will introduce the breaking change, so that by default Notation cli will use referrers API, however, user can choose to use tag schema even if registries support referrers API by setting `--force-referrers-tag` to `true`

> In Feb 2024, the Open Container Initiative (OCI) community released version 1.1.0, which includes the [OCI image specification v1.1.0](https://github.com/opencontainers/image-spec/releases/tag/v1.1.0) and the [OCI distribution specification v1.1.0](https://github.com/opencontainers/distribution-spec/releases/tag/v1.1.0). Notation now adheres to the OCI spec v1.1.0, leading to the deprecation of the experimental flag `--allow-referrers-api`. A new flag, `--force-referrers-tag` (default to `true`), has been introduced to the `notation sign` command, enabling users to choose the referrers tag schema over the [referrers API](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#enabling-the-referrers-api) if the registry they are using does not yet support the referrers API. The `notation verify/list/inspect` commands attempt to use the referrers API first and automatically fall back to the [referrers tag schema](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#referrers-tag-schema) if the referrers API is not supported by the registry.

Reviewer 4: Read this sentence from the blog post:

> The `notation verify/list/inspect` commands attempt to use the referrers API first and automatically fall back to the [referrers tag schema](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#referrers-tag-schema) if the referrers API is not supported by the registry.

This is the proper OCI 1.1 client behavior, i.e., the client must check to see if the registry supports the referrers API before trying to use it and if the registry does not support the referrers API the client must use the referrers tag fallback.

You say:

> If all registries support referrers API, then there is no need for adding this flag `--force-referrers-tag`.

If notation is OCI 1.1 compliant, it does not matter if all registries support the referrers API or not. The _only_ use of the `--force-referrers-tag` command-line option is if the client wants to use the referrers tag fallback _even if a registry supports the referrers API._ If a registry does not support the referrers API, the client _has_ to use the referrers tag fallback, there is no other option. You acknowledge this in this later statement:

> user can choose to use tag schema even if registries support referrers API by setting `--force-referrers-tag` to `true`

Therefore, this statement in the blog is misleading/wrong:

> A new flag, `--force-referrers-tag` (default to `true`), has been introduced to the `notation sign` command, enabling users to choose the referrers tag schema over the [referrers API](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#enabling-the-referrers-api) if the registry they are using does not yet support the referrers API.

The end of the sentence should read "… even if the registry they are using supports the referrers API."

> In Feb 2024, the Open Container Initiative (OCI) community released version 1.1.0, which includes the [OCI image specification v1.1.0](https://github.com/opencontainers/image-spec/releases/tag/v1.1.0) and the [OCI distribution specification v1.1.0](https://github.com/opencontainers/distribution-spec/releases/tag/v1.1.0). Notation now adheres to the OCI spec v1.1.0, leading to the deprecation of the experimental flag `--allow-referrers-api`. A new flag, `--force-referrers-tag` (default to `true`), has been introduced to the `notation sign` command, enabling users to choose the referrers tag schema over the [referrers API](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#enabling-the-referrers-api) if the registry they are using does not yet support the referrers API. The `notation verify/list/inspect` commands attempt to use the referrers API first and automatically fall back to the [referrers tag schema](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#referrers-tag-schema) if the referrers API is not supported by the registry.

Author: @whalelines I think I understood your point now. You mean with the current default behavior, tag schema is used over the referrers API even if the registry they are using supports the referrers API. I will create a PR to amend the blog. Thanks.

> [!NOTE]

> > We will change the default value of `--force-referrers-tag` to `false` in Notation v2.0 release, making referrers API usage as the default.

Author: @whalelines if you mean the note, the breaking changes will be done in Notation v2, the current release v1.2.0 does not introduce any breaking changes.

> > We will change the default value of `--force-referrers-tag` to `false` in Notation v2.0 release, making referrers API usage as the default.

Reviewer 4: I was referring to the removal of the `--allow-referrers-api` command-line option, but it appears that may not have been removed in this release.

> > We will change the default value of `--force-referrers-tag` to `false` in Notation v2.0 release, making referrers API usage as the default.

Author: @whalelines `--allow-referrers-api` is an experimental flag, so removing it is not a breaking change. This change was based on the discussion here https://github.com/notaryproject/notation/issues/892

### Support for RFC 3161 compliant timestamping

Since this release, Notation supports RFC 3161 compliant timestamping. Digital signatures must be generated within the certificate's validity period, as expired certificates compromise the signature's trustworthiness. Timestamping extends the trust of signatures created within certificate validity, allowing successful signature verification even after certificates have expired. Notation's timestamping feature is built on top of the [tspclient-go](https://github.com/notaryproject/tspclient-go) library.

> Learn more at the document [how to sign and verify artifacts in OCI-compliant registries with timestamping](/docs/docs/user-guides/how-to/timestamping/).

Reviewer: ```suggestion
Learn more at the document [how to sign and verify artifacts in OCI-compliant registries with timestamping](/docs/user-guides/how-to/timestamping/).

```

Reviewer: ![image](https://github.com/user-attachments/assets/6b80e14d-f956-45cf-8892-d628c59ba395)

### Other changes

Notation CLI now offers the `armv7` binary, enabling its usage in environments such as Internet of Things (IoT) or embedded systems.

## Get started with Notation v1.2.0

> You can follow the [quick start](/docs/quickstart-guides/quickstart-sign-image-artifact/) to try Notation v1.2.0 for basic signing and verification workflow.

Reviewer 4: ```suggestion
You can follow the [quick start guide](/docs/quickstart-guides/quickstart-sign-image-artifact/) to try Notation v1.2.0 for basic signing and verification workflow.

```

### Support OCI specification v1.1.0

In Feb 2024, the Open Container Initiative (OCI) community released version 1.1.0, which includes the [OCI image specification v1.1.0](https://github.com/opencontainers/image-spec/releases/tag/v1.1.0) and the [OCI distribution specification v1.1.0](https://github.com/opencontainers/distribution-spec/releases/tag/v1.1.0). Notation now adheres to the OCI spec v1.1.0, leading to the deprecation of the experimental flag `--allow-referrers-api`. A new flag, `--force-referrers-tag` (default to `true`), has been introduced to the `notation sign` command, enabling users to choose the referrers tag schema over the [referrers API](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#enabling-the-referrers-api) if the registry they are using does not yet support the referrers API. The `notation verify/list/inspect` commands attempt to use the referrers API first and automatically fall back to the [referrers tag schema](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#referrers-tag-schema) if the referrers API is not supported by the registry.

> [!NOTE]

> > We may change the default value of `--force-referrers-tag` to `false` in Notation v2.0 release, making referrers API usage as the default.

Reviewer 3: do you want to replace `may` to `will`?
